app/routes.py

```diff
@@ -1 +1,164 @@
+    task_response = [task.to_dict() for task in tasks]
+
+    return jsonify(task_response)
+
+
+
+@task_bp.route("/<task_id>", methods=["GET"])
+def get_one_task(task_id):
+    return read_one_model(Task, task_id)
+
+
+@task_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_model(Task, task_id)
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+    task.is_complete = False
+
+    db.session.commit()
+
+    return make_response({"task": task.to_dict()}, 200)
+
+
+@task_bp.route("/<task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    return delete_model(Task, task_id)
+
+@task_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete(task_id):
+    task = validate_model(Task, task_id)
+    task.completed_at = datetime.now()
+
+    db.session.commit()
+
+    post_to_slack(task.title)
+
+    return make_response({"task": task.to_dict()}, 200)
+
+
+@task_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete(task_id):
+    task = validate_model(Task, task_id)
+    task.completed_at = None
+
+    db.session.commit()
+
+    return {"task": task.to_dict()}
+
+
+
+#######GOAL ROUTES ###############
+
+
+@goal_bp.route("", methods=["POST"])
+def create_goal():
+    return create_new_model(Goal, request.get_json())
+
+
+@goal_bp.route("", methods=["GET"])
+def get_goals():
+    goals = Goal.query.all()
+    goals_response = [goal.to_dict() for goal in goals]
+
+    return jsonify(goals_response)
+
+
+
+@goal_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+    return read_one_model(Goal, goal_id)
+
+
+@goal_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    request_body = request.get_json() 
+
+    goal.title = request_body["title"]
+
+    db.session.commit()
+
+    return jsonify({"goal": goal.to_dict()}), 200
+
+
+
+@goal_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+    return delete_model(Goal, goal_id)
+
+
+@goal_bp.route("/<goal_id>/tasks", methods=["POST"])
+def tasks_to_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    request_body = request.get_json()
+
+    goal.tasks = []
+    for task_id in request_body["task_ids"]:
+        task = validate_model(Task, task_id)
+        goal.tasks.append(task)
+
+    db.session.commit()
+
+    return jsonify({
+        "id" : goal.goal_id,
+        "task_ids" : request_body["task_ids"]
+    })
+
+
+
+@goal_bp.route("/<goal_id>/tasks", methods=["GET"])
+def get_tasks_of_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    if not goal:
+        abort(make_response({"message": f"Goal {goal_id} not found"}, 404))
+
+    tasks = []
+    for task in goal.tasks:
+        tasks.append(task.to_dict())
+
+    response = {
+        "id": goal.goal_id,
+        "title": goal.title,
+        "tasks": tasks
+    }
```

Comment:
Like my comment in the tasks_to_goal() method, this code looks really similar to the to_dict goal instance method. Prefer that this logic is in the instance method instead of being repeated here.